src/main/kotlin/net/maizegenetics/phgv2/cli/CreateRefVcf.kt

```diff
@@ -191,15 +198,47 @@ class CreateRefVcf : CliktCommand(help = "Create a haplotype VCF for the referen
             // Load to an hvcf file, write files to user specified outputDir
 
             val hvcfFileName = "${refName}.h.vcf"
-            var localRefHVCFFile = outputDir + "/" + hvcfFileName
+            val tiledbHvcfDir = "${dbPath}/hvcf_files"
+
+            // Make a folder for the hvcf files
+            // ALl hvcf files loaded to this tiledb dataset will be saved in this subfolder
+            Files.createDirectories(Paths.get(tiledbHvcfDir)) // skips creating folders that already exist
+            var localRefHVCFFile = tiledbHvcfDir + "/" + hvcfFileName
 
             //  This is in VariantUtils - it exports the gvcf file.
             // Include the VCF ALT Header lines created in the loop above
             exportVariantContext(refName, fullRefVCList, localRefHVCFFile, myRefSequence!!,altHeaderLines)
             //bgzip and csi index the file
             val bgzippedGVCFFileName = bgzipAndIndexGVCFfile(localRefHVCFFile)
-            myLogger.info("${bgzippedGVCFFileName} created and stored to ${outputDir}")
+            myLogger.info("${bgzippedGVCFFileName} created and stored to ${tiledbHvcfDir}")
+
+            // Create the tiledb datasets if they don't exist
+            File(dbPath).mkdirs()
+            Initdb().createDataSets(dbPath)
+
+            // Load the Ref hvcf file to tiledb
+            val loadVcf = LoadVcf()
+            loadVcf.loadVcfFiles(tiledbHvcfDir,dbPath,"1") // default the threads to "1" for the ref.
+
+            // Store the bed file and the reference fasta file to the dbPath/reference folder
+            // Create the dbPath/reference folder if it doesn't exist
+            val tiledbRefDir = "${dbPath}/reference"
+            Files.createDirectories(Paths.get(tiledbRefDir)) // skips folders that already exist
+            var localBedFile = tiledbRefDir + "/" + Paths.get(ranges).fileName.toString()
```

Comment:
Use a string template
`
val fileName = Paths.get(ranges).fileName
val localBedFile = "${tiledbRefDir}/${fileName}${if(!fileName.endsWith(".bed")) ".bed" else ""}"
`

There might be a function to add the extension if missing already.

Reply:
The problem isn't just a missing extension, it's an extension that isn't ".bed".  User might have called the file "intervals.txt".  Whatever the file is named, I need it to end with .bed as that is what we search for when processing the Variants endpoint.

Reply:
I like your last suggestion - code updated to that and re-tested.